ingestr/src/google_analytics/helpers.py

```diff
@@ -170,3 +222,72 @@ def _resolve_dimension_value(dimension_name: str, dimension_value: str) -> Any:
         return pendulum.from_format(dimension_value, "YYYYMMDDHHmm", tz="UTC")
     else:
         return dimension_value
+
+
+def convert_minutes_ranges_to_minute_range_objects(minutes_ranges: str) -> List[MinuteRange]:
+    minutes_ranges = minutes_ranges.strip()
+    minutes = minutes_ranges.replace(" ", "").split(",")
+
+    if minutes_ranges == "":
+        raise ValueError(
+            "Invalid input. Minutes range should be startminute-endminute format. For example: 1-2,5-6"
+        )
+
+    if "-" not in minutes_ranges:
+        raise ValueError(
+            "Invalid input. Minutes range should be startminute-endminute format. For example: 1-2,5-6"
+        )
+
+    if minutes_ranges.count("-") > 2 or minutes_ranges.count(",") > 1:
+        raise ValueError(
+            "You can define up to two time minutes ranges, formatted as comma-separated values `0-5,25-29`"
+        )
+
+    minute_ranges = []
+    for min in minutes:
+        parts = min.split("-")
+        if len(parts) != 2 or not parts[0] or not parts[1]:
+            raise ValueError(
+                "Invalid input. Minutes range should be startminute-endminute format. For example: 1-2,5-6"
+            )
+        for part in parts:
+            minute_ranges.append(int(part))
```

Comment:
you don't need the second loop below, add them to the minute_ranges directly as objects here.